### PR TITLE
Make a flag to disable "Change tab on scroll events"

### PR DESCRIPTION
**Pull request by sangwoo108**
Resolves https://github.com/brave/brave-browser/issues/27694

On linux, users can switch the active tab with wheel scroll. But for some users, it feels annoying. so it'd be good to have a flag to disable that.

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### Manual
When the flag("Change the active tab on scroll events") is disabled, scrolling wheel on tab strip shouldn't change the active tab.

### CODE REVIEW

File: browser/ui/views/tabs/features.cc [hunk 1]
```diff
@@ -30,6 +30,14 @@ BASE_FEATURE(kBraveVerticalTabs,
              "BraveVerticalTabs",
              base::FEATURE_DISABLED_BY_DEFAULT);
 
+BASE_FEATURE(kBraveChangeActiveTabOnScrollEvent,
+             "BraveChangeActiveTabOnScrollEvent",
+#if BUILDFLAG(IS_LINUX)
+             base::FEATURE_ENABLED_BY_DEFAULT);
+#else
+             base::FEATURE_DISABLED_BY_DEFAULT);
```
Comment (SergeyZhukovsky) [inline browser/ui/views/tabs/features.cc:38]: shouldn't it be just enabled for any OS?

Comment (sangwoo108) [reply browser/ui/views/tabs/features.cc:38]: I'd like to have it mirror browser_defaults::kScrollEventChangesTab

File: browser/ui/views/frame/brave_browser_root_view.cc [hunk 2]
```diff
@@ -16,6 +16,11 @@ BraveBrowserRootView::~BraveBrowserRootView() = default;
 
 bool BraveBrowserRootView::OnMouseWheel(const ui::MouseWheelEvent& event) {
   // Bypass BrowserRootView::OnMouseWheel() to avoid tab cycling feature.
+  if (!base::FeatureList::IsEnabled(
+          tabs::features::kBraveChangeActiveTabOnScrollEvent)) {
```
[the remark this answers is not in the document]
Comment (sangwoo108) [reply browser/ui/views/frame/brave_browser_root_view.cc:20]: Yeah, I think we can add Linux guard for it.

**Comment by sangwoo108** [comment]
@SergeyZhukovsky I've updated guards so that new code would be included only when it's Linux. Please take a look again :)

**Review by SergeyZhukovsky — APPROVED** [review]
++

**Comment by sangwoo108** [comment]
@SergeyZhukovsky Thanks for the review!

**Pull request merged**